security/pkg/credentialfetcher/plugin/mockintetestcredfetcher.go

```diff
@@ -0,0 +1,39 @@
+package plugin
+
+import (
+	"istio.io/istio/pkg/security"
+	"istio.io/pkg/log"
+)
+
+var (
+	mockInteTestPluginLog = log.RegisterScope("mockintetestplugin", "Mock Integration credential fetcher for istio agent", 0)
+)
+
+// The plugin object.
+type MockInteTestPlugin struct {
+	token string
+}
+
+// CreateMockPlugin creates a mock credential fetcher plugin. Return the pointer to the created plugin.
+func CreateMockInteTestPlugin(token string) *MockInteTestPlugin {
```

Comment:
We don't need the mockintetestcredfetcher.go, just combine the logic into plugin/mock.go. e.g. 
func CreateMockPlugin(token string) *MockPlugin {
        t := "test_token"
        if token != nil {
            t = token
        }
	p := &MockPlugin{
              token: t ,
        }	
        return p
}